setup-env

```diff
@@ -0,0 +1,175 @@
+#!/usr/bin/env bash
+
+set -o nounset
+set -o errexit
+set -o pipefail
+
+USAGE=$(cat << 'END_OF_LINE'
+This script is used to configure a developement environment for this repo.
+
+It does the following:
+  - Verifies pyenv and pyenv-virtualenv are installed.
+  - Creates a Python virtual environment.
+  - Configures the activation of the virtual enviroment for the repo directory.
+  - Installs the requirements required for development.
+  - Installs git pre-commit hooks.
+  - Configures git upstream remote "lineage" repositories.
+
+usage: setup-env [--force] [--help] [virt_env_name]
+
+END_OF_LINE
+)
+
+# Flag to force deletion and creation of virtual environment
+FORCE=0
+
+# Positional parameters
+PARAMS=""
+
+# Parse command line arguments
+while (( "$#" )); do
+  case "$1" in
+    -f|--force)
+      FORCE=1
+      shift
+      ;;
+    -h|--help)
+      echo "${USAGE}"
+      exit 0
+      ;;
+    -*) # unsupported flags
+       echo "Error: Unsupported flag $1" >&2
+       exit 1
+       ;;
+    *) # preserve positional arguments
+       PARAMS="$PARAMS $1"
+       shift
+       ;;
+      esac
+done
+
+# set positional arguments in their proper place
+eval set -- "$PARAMS"
+
+# Check to see if pyenv is installed
+if [ -z "$(which pyenv)" ] || [ -z "$(which pyenv-virtualenv)" ]; then
+  echo "pyenv and pyenv-virtualenv are required."
+  if [[ "$OSTYPE" == "darwin"* ]]; then
+    cat << 'END_OF_LINE'
+
+  On the Mac, installation is as simple as "brew install pyenv
```

Comment:
This sentence assumes that `brew` has already been installed.

Reply:
Most of the text is copied directly from `CONTRIBUTING.md`.  
See: https://github.com/cisagov/skeleton-generic/blame/develop/CONTRIBUTING.md#L51

Reply:
> Most of the text is copied directly from `CONTRIBUTING.md`.
> See: https://github.com/cisagov/skeleton-generic/blame/develop/CONTRIBUTING.md#L51

If we change one we should change both.

Reply:
Both changed. 
See: 996ae26